helix-core/src/main/java/org/apache/helix/monitoring/mbeans/ParticipantStatusMonitor.java

```diff
@@ -115,42 +122,36 @@ public void reportTransitionStat(StateTransitionContext cxt, StateTransitionData
   }
 
   private ObjectName getObjectName(String name) throws MalformedObjectNameException {
-    return new ObjectName(String.format("%s:%s", MonitorDomainNames.CLMParticipantReport.name(), name));
+    return new ObjectName(
+        String.format("%s:%s", MonitorDomainNames.CLMParticipantReport.name(), name));
   }
 
-  private void register(Object bean, ObjectName name) {
-    LOG.info("Registering bean: " + name.toString());
-    if (_beanServer == null) {
-      LOG.warn("bean server is null, skip reporting");
-      return;
-    }
-    try {
-      _beanServer.unregisterMBean(name);
-    } catch (Exception e1) {
-      // Swallow silently
-    }
-
-    try {
-      _beanServer.registerMBean(bean, name);
-    } catch (Exception e) {
-      LOG.warn("Could not register MBean", e);
-    }
+  /**
+   * Build participant bean name
+   * @param participantName
+   * @return participant bean name
+   */
+  protected String getParticipantBeanName(String participantName) {
+    return String.format("%s=%s", PARTICIPANT_KEY, participantName);
   }
 
   public void shutDown() {
     if (_messageMonitor != null) {  // is participant
```

Comment:
These are not necessary anymore, right? Since you added the unregister() call below for _messageMonitor.